src/NServiceBus.AwsLambda.SQS/AwsLambdaSQSEndpointConfiguration.cs

```diff
@@ -21,27 +21,47 @@ public class AwsLambdaSQSEndpointConfiguration
         /// </summary>
         /// <param name="endpointName">The endpoint name to be used.</param>
         public AwsLambdaSQSEndpointConfiguration(string endpointName)
+            : this(endpointName, null, null)
         {
-            EndpointConfiguration = new EndpointConfiguration(endpointName);
 
-            EndpointConfiguration.UsePersistence<InMemoryPersistence>();
+        }
+
+        /// <summary>
+        /// Creates a serverless NServiceBus endpoint running with an AmazonSQS SQS trigger.
+        /// </summary>
+        public AwsLambdaSQSEndpointConfiguration(string endpointName, IAmazonSQS sqsClient, IAmazonSimpleNotificationService snsClient)
+        {
+            EndpointConfiguration = new EndpointConfiguration(endpointName);
 
             LogManager.Use<LambdaLoggerDefinition>();
 
-            //make sure a call to "onError" will move the message to the error queue.
-            EndpointConfiguration.Recoverability().Delayed(c => c.NumberOfRetries(0));
-            // send failed messages to the error queue
             recoverabilityPolicy.SendFailedMessagesToErrorQueue = true;
-            EndpointConfiguration.Recoverability().CustomPolicy(recoverabilityPolicy.Invoke);
 
-            Transport = UseTransport<SqsTransport>();
+            //make sure a call to "onError" will move the message to the error queue.
+            EndpointConfiguration.Recoverability()
+                .Delayed(c => c.NumberOfRetries(0))
+                .CustomPolicy(recoverabilityPolicy.Invoke);
+
+            if (sqsClient is null && snsClient is null)
+            {
+                // If both sqsClient/snsClient are null, use default constructor which will instantiate a default client instance.
+                Transport = new SqsTransport();
+            }
+            else
+            {
+                // If either sqsClient or snsClient is null the transport will throw.
```

Comment:
isn't that what the test setup (at the very beginning of this PR) does?

Reply:
I fixed the tests to pass an SNS Client.  I thought the transport threw if the client is null, but after further review,  the SqsTransport does not throw.  Though in my opinion, the behavior of the transport on this is a bit unpredictable.  If you call the default constructor it will initialize them to default implementations, if you call this other one, it does not check for null nor does it initialize default implementations.  I'm not sure if sending null in for these has a predictable outcome and fear it's a bit of a time bomb for when you find out if sending in null is an ok scenario.